### PR TITLE
mgr/dashboard: Predefine labels in create host form

### PR DESCRIPTION
**Pull request by nizamial09**
Also retains the previously created labels by user in the form

Screenshot from 2021-11-18 16-44-05

Fixes: https://tracker.ceph.com/issues/53315
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [x] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
I couldn't figure out a way to include unit tests but I wrote the e2es.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.ts [hunk 1]
```diff
@@ -21,6 +21,8 @@ export class HostService extends ApiClient {
   baseURL = 'api/host';
   baseUIURL = 'ui-api/host';
 
+  predefinedLabels = ['mon', 'mgr', 'osd', 'mds', 'rgw', 'nfs', 'iSCSI', 'rbd', 'grafana'];
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.ts:24]: Everything lowercase?
```suggestion
  predefinedLabels = ['mon', 'mgr', 'osd', 'mds', 'rgw', 'nfs', 'iscsi', 'rbd', 'grafana'];
```

**Review by epuertat — APPROVED** [review]

**Comment by epuertat** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.ts (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.ts]: done! Thanks @epuertat

**Review by pereman2 — APPROVED** [review]

**Review by avanthakkar — APPROVED** [review]

**Comment by avanthakkar** [comment]
> I couldn't figure out a way to include unit tests but I wrote the e2es.

Maybe by faking `predefinedLabels` from service (with some duplicates maybe) and checking if `uniqueLabels` got all unique labels?

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
@nizamial09 I'm seeing [regular/constant failures ](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/1466/consoleFull#-195686988373a8703b-5adb-41c5-84a0-8cf4e065ba3d) in the cephadm run. Could you please check what's going on there?

**Comment by nizamial09** [comment]
> @nizamial09 I'm seeing [regular/constant failures ](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/1466/consoleFull#-195686988373a8703b-5adb-41c5-84a0-8cf4e065ba3d) in the cephadm run. Could you please check what's going on there?

@epuertat this PR from @alfonsomthd is trying to address those. https://github.com/ceph/ceph/pull/44045

**Pull request merged**